hypothesis-python/src/hypothesis/strategies/_internal/strategies.py

```diff
@@ -554,7 +556,13 @@ def do_draw(self, data):
         return result
 
     def get_element(self, i):
-        return self._transform(self.elements[i])
+        res = self.elements[i]
+        if self.reference_to_value:
+            value = self.reference_to_value(res)
+            if self._transform(value) is filter_not_satisfied:
+                return filter_not_satisfied
+            return res
+        return self._transform(res)
```

Comment:
I think we still want to `self._transform` on line 564, right?  So something like

```suggestion
        value = self.elements[i]
        if self.reference_to_value:
            value = self.reference_to_value(value)
        return self._transform(value)
```